### PR TITLE
Add training API

### PR DESCRIPTION
**Pull request by vvmnnnkv**
## Description
This PR implements changes described in #212 

## Affected Dependencies
n/a

## How has this been tested?
Added unit tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: examples/mnist/index.js [hunk 1]
```diff
@@ -24,111 +24,74 @@ startButton.onclick = () => {
...
+    // Shuffle dataset.
+    // TODO replace with Dataloader API
+    updateStatus('Shuffling MNIST data...');
+    const indices = Array.from(tf.util.createShuffledIndices(trainDataset.xs.shape[0]));
+    const data = trainDataset.xs.gather(indices);
+    const target = trainDataset.labels.gather(indices);
+    updateStatus('MNIST data shuffled.');
+
+    const training = job.train('training_plan', {
+      planInputs: [
```
Comment (cereallarceny) [inline examples/mnist/index.js]: Can we just call these `inputs`?

File: examples/mnist/index.js [hunk 2]
```diff
@@ -24,111 +24,74 @@ startButton.onclick = () => {
...
+        new PlanInputSpec(PlanInputSpec.TYPE_DATA),
+        new PlanInputSpec(PlanInputSpec.TYPE_TARGET),
+        new PlanInputSpec(PlanInputSpec.TYPE_BATCH_SIZE),
+        new PlanInputSpec(PlanInputSpec.TYPE_VALUE, 'lr', null, clientConfig.lr),
+        new PlanInputSpec(PlanInputSpec.TYPE_MODEL_PARAM, 'W1', 0),
+        new PlanInputSpec(PlanInputSpec.TYPE_MODEL_PARAM, 'b1', 1),
+        new PlanInputSpec(PlanInputSpec.TYPE_MODEL_PARAM, 'W2', 2),
+        new PlanInputSpec(PlanInputSpec.TYPE_MODEL_PARAM, 'b2', 3),
+      ],
+      planOutputs: [
```
Comment (cereallarceny) [inline examples/mnist/index.js]: Can we just call these `outputs`?

**Review by cereallarceny — CHANGES_REQUESTED** [review]
Only minor changes. However @vvmnnnkv - I will no longer merge PR's without appropriate documentation changes. Please also supply changes in documentation to all the appropriate readmes both in this project and in others.

**Comment by vvmnnnkv** [comment]
Updated docs and added additional input spec type to easier reference values in PyGrid's client_config.

**Review by cereallarceny — APPROVED** [review]

**Pull request merged**